zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java

```diff
@@ -297,11 +298,13 @@ public void emptyTrash(AuthenticationInfo subject) throws IOException {
   public void restoreAll(AuthenticationInfo subject) throws IOException {
     NoteManager.Folder trash = noteManager.getTrashFolder();
     // restore notes under trash folder
-    for (NoteManager.NoteNode noteNode : trash.getNotes().values()) {
+    List<NoteNode> notes = trash.getNotes().values().stream().collect(Collectors.toList());
```

Comment:
Could you add one line of comment why we need to do like this ? So that others won't refactor it in future

Reply:
I'll add a comment